common/src/primitives/semver.rs

```diff
@@ -15,15 +15,15 @@
 
 use serialization::{Decode, Encode};
 
-#[derive(Debug, PartialEq, Eq, Encode, Decode, Copy, Clone)]
+#[derive(Debug, PartialEq, Eq, Encode, Decode, Copy, Clone, Hash)]
```

Comment:
Replace this with Ord and PartialOrd. At least, having a correct and defined order for SemVer is a win anyway.

Reply:
Btw, from what I'm seeing in the implementation, the ordering will be correct by just deriving. Please add a test or two to ensure that this will remain to be the case.